### PR TITLE
Stlx 13809 implement is wallet address for algo

### PR DESCRIPTION
**Pull request by charly-simtlix**
Companion Story for https://bitgoinc.atlassian.net/browse/BG-43270

**NOTE**: Please take a look if the path taken was correct

**NTE**: This PR requires https://github.com/BitGo/bitgo-microservices/pull/16489 to be merged first

### CODE REVIEW

File: modules/core/src/v2/coins/algo.ts [hunk 1]
```diff
@@ -441,8 +447,33 @@ export class Algo extends BaseCoin {
...
 
-  isWalletAddress(): boolean {
-    throw new MethodNotImplementedError();
+  /**
+   * Check if address can be used to send funds.
+   * Only root address is allowed to do that.
+   * @param params.address address to validate
+   * @param params.keychains public keys to generate the wallet
+   */
+  isWalletAddress(params: VerifyAddressOptions): boolean {
```
Comment (lcovar) [inline modules/core/src/v2/coins/algo.ts:456]: @OttoAllmendinger can you take a look at this implementation?

File: modules/core/src/v2/coins/algo.ts [hunk 2]
```diff
@@ -441,8 +464,38 @@ export class Algo extends BaseCoin {
     return {};
   }
 
-  isWalletAddress(): boolean {
-    throw new MethodNotImplementedError();
+  /**
+   * Check if address can be used to send funds.
+   * Only root address is allowed to do that.
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/algo.ts:469]: is this still accurate?

File: modules/core/src/v2/coins/algo.ts [hunk 3]
```diff
@@ -441,8 +464,38 @@ export class Algo extends BaseCoin {
...
+   */
+  isWalletAddress(params: VerifyAlgoAddressOptions): boolean {
+    const {
+      address,
+      keychains,
+      coinSpecific: { bitgoPubKey },
+    } = params;
+
+    if (!this.isValidAddress(address)) {
+      // TODO Should throw error or simple return false?
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/coins/algo.ts:481]: better throw an error here

**Review by ofornaroli — APPROVED** [review]
LGTM

**Review by lcovar — APPROVED** [review]
overall lgtm.

lets get @OttoAllmendinger blessing here

File: modules/core/src/v2/coins/algo.ts [hunk 4]
```diff
@@ -453,7 +505,38 @@ export class Algo extends BaseCoin {
...
+   * Stellar and Algorand both use keys on the ed25519 curve, but use different encodings.
+   * As the HSM doesn't have explicit support to create Algorand addresses, we use the Stellar
+   * keys and re-encode them to the Algorand encoding.
+   *
+   * This method should only be used when creating Algorand custodial wallets reusing Stellar keys.
+   *
+   * @param {string} addressOrPubKey a Stellar pubkey or Algorand address
+   * @return {*}
+   */
+  private stellarAddressToAlgoAddress(addressOrPubKey: string): string {
```
Comment (mmcshinsky-bitgo) [inline modules/core/src/v2/coins/algo.ts:522]: Not a blocker but will there be a needed use case for `algoAddressToStellarAddress` for sending data back through our service layer in the accepted Stellar format?

**Pull request merged**